scipy/optimize/_shgo.py

```diff
@@ -682,7 +683,14 @@ def init_options(self, options):
         None
 
         """
+        # Ensure that 'jac' and 'hess' are passed directly to `minimize` as
+        # keywords, not as part of its 'options' dictionary
+        self.minimizer_kwargs['jac'] = options.pop('jac', None)
+        self.minimizer_kwargs['hess'] = options.pop('hess', None)
```

Comment:
I tried to simplify this, but I see that this could replace `minimizer_kwargs['jac']` with `None` if the user passes the `jac` to `minimizer_kwargs` but not to `shgo`'s `options`. But is there any good reason that the user would want the local minimizer to have the gradient but not let `shgo` use it? If `jac` is found in `minimizer_kwargs`, shouldn't it go to `sgho`, too?